code/obj/item/device/transfer_valve.dm

```diff
@@ -518,45 +518,103 @@ TYPEINFO(/obj/item/device/transfer_valve/briefcase)
 		else
-			if(crystal.pressure)
-				boutput( user, "The reader reads <b>[crystal.pressure/25]</b> kilojoules." )
-			else
-				boutput( user, "The reader reads a firm 0. It guilts you into trying to read an unexploded pressure crystal, and seems to have succeeded. You feel ashamed for being so compelled by a device that has nothing more than a slot and a number display.")
+			boutput( user, "<span class='notice'>The display reads a firm 0. It guilts you into trying to read an unexploded pressure crystal, \
+							and seems to have succeeded. You feel ashamed for being so compelled by a device that \
+							has nothing more than a slot and a number display.</span>")
+
 	ex_act()
+		if (src.crystal)
+			src.crystal.set_loc(src.loc)
 		qdel(src)
+
 	attackby(obj/item/thing, mob/user)
-		if(istype(thing, /obj/item/pressure_crystal))
-			if(src.crystal)
-				boutput( user, "You contemplate how to place the crystal in an occupied sensor, but can't manage to figure out how." )
-			else
-				src.crystal = thing
-				thing.pixel_x = 0
-				thing.pixel_y = 0
-				boutput( user, "You insert the crystal." )
-				overlays += thing
-				user.drop_item()
-				crystal.set_loc(src)
-				wear_image.overlays += src.crystal
+		if(istype(thing, /obj/item/pressure_crystal) && src.insert_crystal(user, thing))
+			user.visible_message("[user] inserts [thing] into [src].",
+								"<span class='notice'>You insert the crystal into [src].</span>")
+		else if(src.crystal && thing.tool_flags & TOOL_PRYING && src.remove_crystal(user))
+			user.visible_message("[user] pries the crystal from [src].",
+								"<span class='notice'>You pry out [src]'s crystal.</span>")
+		else return ..()
+
+	mouse_drop(atom/over_object, src_location, over_location)
+		if (!src.crystal)
+			. = ..()
 			return
-		else if(thing.tool_flags & TOOL_PRYING && src.crystal)
-			overlays = list()
-			wear_image.overlays = list()
-			boutput( user, "You pry out the crystal." )
-			if(prob(src.crystal.total_pressure / 45))
-				boutput( user, "<b class='alert'>It shatters!</b>" )
-				qdel(src.crystal)
-				return
-			src.crystal.set_loc(user.loc)
-			src.crystal = null
+		if(BOUNDS_DIST(src_location, usr) > 0 || BOUNDS_DIST(src_location, over_location) > 0)
 			return
-		else return ..()
+		if (src.remove_crystal(usr, over_location))
+			usr.visible_message("[usr] removes the crystal from [src].",
+								"<span class='notice'>You pull the crystal out of [src].</span>")
+
+	MouseDrop_T(atom/movable/thing, mob/user)
+		. = ..()
+		if (!istype(thing, /obj/item/pressure_crystal))
+			return
+		if (src.insert_crystal(user, thing))
+			user.visible_message("[user] drops [thing] into [src].",
+								"<span class='notice'>You drop the crystal into [src].</span>")
+
+	attack_hand(mob/user)
+		if (src.loc != user || !user.find_in_hand(src))
+			..()
+		else if (src.crystal && src.remove_crystal(user))
```

Comment:
This removes the crowbar requirement, either make it require a crowbar to extract or remove the crowbar interaction entirely.